plugins/main/public/components/common/data-source/pattern/pattern-data-source-filter-manager.ts

```diff
@@ -514,27 +517,42 @@ export class PatternDataSourceFilterManager
           true,
         );
       default:
-        return this.generateFilter(
-          key,
-          value,
-          this.dataSource.id,
-          undefined,
-          controlledBy,
-        );
+        throw new Error('Invalid filter type');
     }
   }
 
+  /**
+   * Returns a filter with the field and value received
+   * @param field
+   * @param value
+   * @returns
+   */
+  createFilter(
+    type: keyof typeof FILTER_OPERATOR,
+    key: string,
+    value: string | [],
+    controlledBy?: string,
+  ): tFilter {
+    return PatternDataSourceFilterManager.createFilter(
+      type,
+      key,
+      value,
+      this.dataSource.id,
+      controlledBy,
+    );
+  }
+
   /**
    * Return a simple filter object with the field, value and index pattern received
```

Comment:
The name of the received parameters should be changed from `indexPatternTitle` to `indexPatternId`.